Model/Feed.php

```diff
@@ -386,16 +476,208 @@ function UserFeed($progressFileName, $storeId, $doNdjson)
+     * sends first row of CSV data, otherwise sends opening string of json.
+     * @param $feedType string One of the Feed::FEED_TYPE_... constants
+     */
+    protected function start($feedType)
+    {
+        if ($feedType == self::FEED_TYPE_ORDER) {
+            $startJson = "OrderId,UserId,Email,DateTimeStamp,ProdCode,Quantity,UnityPrice,LinePrice" . PHP_EOL;
+        } else {
+            $startJson = '{"Version": 2';
+        }
+        $parameters = $this->getParameters($startJson, $feedType);
+        $this->send("feed-create", $parameters);
+        $this->logger->debug("PureClarity: Started feed");
+    }
+
+    /**
+     * End the feed by sending any closing data to the feed-close end point. For order feeds,
+     * no closing data is sent, the end point is simply called. For others, it's simply a closing
+     * bracket.
+     * @param $feedType string One of the Feed::FEED_TYPE_... constants
+     */
+    protected function end($feedType)
+    {
+        $data = ( $feedType == self::FEED_TYPE_ORDER ? '' : '}' );
+        $this->send("feed-close", $this->getParameters($data, $feedType));
+        // Ensure progress file is set to complete
+        $this->coreHelper->setProgressFile($this->progressFileName, 'N/A', 1, 1, "true", "false");
+    }
+
+    /**
+     * Sends the data to the specified end point, i.e. sends feed to PureClarity
+     * @param $endPoint string
+     * @param $parameters array
+     */
+    protected function send($endPoint, $parameters)
+    {
+        
+        $url = $this->coreHelper->getFeedBaseUrl($this->storeId) . $endPoint;
+        
+        $this->logger->debug("PureClarity: About to send data to {$url}: " . print_r($parameters, true));
+
+        $post_fields = http_build_query($parameters);
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_URL, $url);
+        curl_setopt($ch, CURLOPT_CONNECTTIMEOUT_MS, 5000);
+        curl_setopt($ch, CURLOPT_TIMEOUT_MS, 10000);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        curl_setopt($ch, CURLOPT_FOLLOWLOCATION, true);
+        curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
+        curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, 0);
+
+        if (! empty($post_fields)) {
+            curl_setopt($ch, CURLOPT_POST, true);
+            curl_setopt($ch, CURLOPT_POSTFIELDS, $post_fields);
+            curl_setopt($ch, CURLOPT_HTTPHEADER, [
+                    'Content-Type: application/x-www-form-urlencoded',
+                    'Content-Length: ' . strlen($post_fields)
+                ]);
+        } else {
+            curl_setopt($ch, CURLOPT_POST, false);
+        }
+
+        curl_setopt($ch, CURLOPT_FAILONERROR, true);
+        curl_setopt($ch, CURLOPT_VERBOSE, true);
+
+        $response = curl_exec($ch);
+
+        if (curl_errno($ch)) {
+            $this->logger->debug('PureClarity: Error: ' . curl_error($ch));
+            $this->problemFeeds[] = $parameters['feedName'];
```

Comment:
If an error does occur during attempted send, name of feed added to problemFeeds array, and script simply continues. Once all feeds have been sent, the problemFeeds array is then examined by call to checkSuccess() function and any errors reported to user.